examples/DotNetCore/ConsoleAppWithTableStorageReference/Program.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Azure.Data.AppConfiguration;
+using Azure.Data.Tables;
+using Azure.Identity;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Examples.ConsoleAppWithTableStorageReference
+{
+    class Program
+    {
+        static IConfiguration Configuration { get; set; }
+
+        static void Main(string[] args)
+        {
+            Configure();
+
+            StringBuilder sb = BuildProductTable();
+
+            sb.AppendLine();
+
+            sb.AppendLine("Press any key to exit...");
+
+            Console.Clear();
+
+            Console.Write(sb.ToString());            
+
+            // Finish on key press
+            Console.ReadKey();
+        }
+
+        private static void Configure()
+        {
+            var builder = new ConfigurationBuilder();
+
+            builder.AddJsonFile("appsettings.json")
+                    .AddEnvironmentVariables();
+
+            IConfiguration configuration = builder.Build();
+
+            if (string.IsNullOrEmpty(configuration["ConnectionString"]))
+            {
+                Console.WriteLine("Connection string not found.");
+                
+                Console.WriteLine("Please set the 'ConnectionString' environment variable to a valid Azure App Configuration connection string and re-run this example.");
+                
+                return;
+            }
+
+            // Augment the configuration builder with Azure App Configuration
+            // Pull the connection string from an environment variable
+            builder.AddAzureAppConfiguration(options =>
+            {
+                options.Connect(configuration["ConnectionString"])
+                        .Select("*")
+                        .Map((setting) =>
```

Comment:
I'm suprised this isn't async given that we're reading from db.

Reply:
I noticed that in the [documentation](https://learn.microsoft.com/en-us/azure/cosmos-db/table/quickstart-dotnet?toc=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fstorage%2Ftables%2Ftoc.json&bc=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fbread%2Ftoc.json&tabs=azure-cli%2Cwindows#query-items) they use `Query` which isn't async, but not sure why. It looks like there's also a `QueryAsync` that we could use though.

Reply:
Please do.

Reply:
I misunderstood what `QueryAsync` was since it returns an `AsyncPageable` and not a task, which means I can't await it directly.

Reply:
AsyncPageable can be converted to IEnumerable in an async manner though.

Reply:
I think I was missing the point of making it async, fixed it.